Goal: make message from the bot sticky to the bottom of the thread

Problem: currently, if you do not submit the time right away, the message can be lost as new messages from people arrive in the thread.

# Solution
Make the message from the bot always stick to the bottom of the thread (same as conventional commit Action behavior)

```$ npm view @holdex/autoinvoicing@0.0.9
npm ERR! code E403
npm ERR! 403 403 Forbidden - GET https://npm.pkg.github.com/@holdex%2fautoinvoicing - Permission permission_denied: read_package
npm ERR! 403 In most cases, you or one of your dependencies are requesting
npm ERR! 403 a package version that is forbidden by your security policy, or
npm ERR! 403 on a server you do not have access to.

npm ERR! A complete log of this run can be found in: C:\Users\Santa\AppData\Local\npm-cache\_logs\2024-08-08T16_01_38_187Z-debug-0.log```

--- a/integrations/autoinvoicing/package.json
+++ b/integrations/autoinvoicing/package.json
@@ -1,6 +1,6 @@
 {
   "name": "@holdex/autoinvoicing",
-  "version": "0.0.9",
+  "version": "0.0.10",
   "description": "The Autoinvoicing GitHub integration for Trigger.dev",
   "main": "./dist/index.js",
   "types": "./dist/index.d.ts",

--- a/integrations/autoinvoicing/src/index.ts
+++ b/integrations/autoinvoicing/src/index.ts
@@ -301,12 +301,12 @@ const onIssueAssigned: EventSpecification<IssuesAssignedEvent> = {
 };
 
 const onIssueComment: EventSpecification<IssueCommentEvent> = {
-  name: 'issues',
+  name: 'issue_comment',
   title: 'On issue comment',
   source: 'github.com',
   icon: 'github',
   filter: {
-    action: ['commented']
+    action: ['created']
   },
   examples: [issueCommentCreated],
   parsePayload: (payload) => payload as IssueCommentEvent,

--- a/package.json
+++ b/package.json
@@ -21,7 +21,7 @@
     "@discordjs/core": "^1.2.0",
     "@discordjs/rest": "^2.3.0",
     "@google-cloud/bigquery": "^7.3.0",
-    "@holdex/autoinvoicing": "0.0.9",
+    "@holdex/autoinvoicing": "0.0.10",
     "@octokit/graphql-schema": "^15.5.0",
     "@octokit/oauth-methods": "^4.1.0",
     "@octokit/webhooks-types": "^7.5.0",

--- a/src/lib/server/trigger-dev/jobs/check-run.ts
+++ b/src/lib/server/trigger-dev/jobs/check-run.ts
@@ -20,7 +20,9 @@ import {
   getSubmissionStatus,
   submissionCheckPrefix,
   githubApp,
-  bugCheckPrefix
+  bugCheckPrefix,
+  submissionHeaderCommentForPR,
+  bodyWithHeaderForPR
 } from '../utils';
 
 export async function createJob<T extends IOWithIntegrations<{ github: Autoinvoicing }>>(
@@ -258,7 +260,7 @@ async function runSubmissionJob<T extends IOWithIntegrations<{ github: Autoinvoi
     const previous = await getPreviousComment<typeof octokit>(
       { owner: payload.organization, repo: repoDetails.data.name },
       payload.prNumber,
-      submissionHeaderComment(payload.prId.toString()),
+      submissionHeaderCommentForPR(payload.prId.toString()),
       octokit
     );
     return previous;
@@ -268,7 +270,7 @@ async function runSubmissionJob<T extends IOWithIntegrations<{ github: Autoinvoi
   const submissionCreated = result.checkRun?.conclusion === 'SUCCESS';
   let members: string[] = [];
 
-  const commentBody = bodyWithHeader(
+  const commentBody = bodyWithHeaderForPR(
     `<members>⚠️⚠️⚠️\nYou must [submit the time](https://pr-time-tracker.vercel.app/prs/${payload.organization}/${repoDetails.data.name}/${payload.prId}) spent on this PR.\n⚠️⚠️⚠️`,
     payload.prId.toString()
   );
@@ -573,14 +575,6 @@ async function updateCheckRun<T extends Octokit>(octokit: T, input: UpdateCheckR
       `,
     { input }
   );
-}
-
-function submissionHeaderComment(header: string): string {
-  return `<!-- Sticky Pull Request Comment${header} -->`;
-}
-
-function bodyWithHeader(body: string, header: string): string {
-  return `${body}\n${submissionHeaderComment(header)}`;
 }
 
 const regex = new RegExp(/\B@([a-z0-9](?:-(?=[a-z0-9])|[a-z0-9]){0,38}(?<=[a-z0-9]))/, 'gmi');

--- a/src/lib/server/trigger-dev/jobs/check-run.ts
+++ b/src/lib/server/trigger-dev/jobs/check-run.ts
@@ -21,8 +21,8 @@ import {
   submissionCheckPrefix,
   githubApp,
   bugCheckPrefix,
-  submissionHeaderCommentForPR,
-  bodyWithHeaderForPR
+  submissionHeaderCommentForPr,
+  bodyWithHeaderForPr
 } from '../utils';
 
 export async function createJob<T extends IOWithIntegrations<{ github: Autoinvoicing }>>(
@@ -260,7 +260,7 @@ async function runSubmissionJob<T extends IOWithIntegrations<{ github: Autoinvoi
     const previous = await getPreviousComment<typeof octokit>(
       { owner: payload.organization, repo: repoDetails.data.name },
       payload.prNumber,
-      submissionHeaderCommentForPR(payload.prId.toString()),
+      submissionHeaderCommentForPr(payload.prId.toString()),
       octokit
     );
     return previous;
@@ -270,7 +270,7 @@ async function runSubmissionJob<T extends IOWithIntegrations<{ github: Autoinvoi
   const submissionCreated = result.checkRun?.conclusion === 'SUCCESS';
   let members: string[] = [];
 
-  const commentBody = bodyWithHeaderForPR(
+  const commentBody = bodyWithHeaderForPr(
     `<members>⚠️⚠️⚠️\nYou must [submit the time](https://pr-time-tracker.vercel.app/prs/${payload.organization}/${repoDetails.data.name}/${payload.prId}) spent on this PR.\n⚠️⚠️⚠️`,
     payload.prId.toString()
   );

--- a/src/lib/server/trigger-dev/jobs/check-run.ts
+++ b/src/lib/server/trigger-dev/jobs/check-run.ts
@@ -21,8 +21,8 @@ import {
   submissionCheckPrefix,
   githubApp,
   bugCheckPrefix,
-  submissionHeaderCommentForPr,
-  bodyWithHeaderForPr
+  submissionHeaderComment,
+  bodyWithHeader
 } from '../utils';
 
 export async function createJob<T extends IOWithIntegrations<{ github: Autoinvoicing }>>(
@@ -260,7 +260,7 @@ async function runSubmissionJob<T extends IOWithIntegrations<{ github: Autoinvoi
     const previous = await getPreviousComment<typeof octokit>(
       { owner: payload.organization, repo: repoDetails.data.name },
       payload.prNumber,
-      submissionHeaderCommentForPr(payload.prId.toString()),
+      submissionHeaderComment('Pull Request', payload.prId.toString()),
       octokit
     );
     return previous;
@@ -270,7 +270,8 @@ async function runSubmissionJob<T extends IOWithIntegrations<{ github: Autoinvoi
   const submissionCreated = result.checkRun?.conclusion === 'SUCCESS';
   let members: string[] = [];
 
-  const commentBody = bodyWithHeaderForPr(
+  const commentBody = bodyWithHeader(
+    'Pull Request',
     `<members>⚠️⚠️⚠️\nYou must [submit the time](https://pr-time-tracker.vercel.app/prs/${payload.organization}/${repoDetails.data.name}/${payload.prId}) spent on this PR.\n⚠️⚠️⚠️`,
     payload.prId.toString()
   );

--- a/src/lib/server/trigger-dev/jobs/index.ts
+++ b/src/lib/server/trigger-dev/jobs/index.ts
@@ -10,6 +10,7 @@ import { createJob as createPrJob } from './pull-requests';
 import { createJob as createPrReviewJob } from './pull-requests-review';
 import { createJob as createIssueJob } from './issues';
 import { createJob as createIssueCreationJob } from './issues-creation';
+import { createJob as createIssueCommentJob } from './issues-comment';
 import { createJob as createCheckRunJob, createEventJob as createCheckEventJob } from './check-run';
 
 config.integrationsList.forEach((org) => {
@@ -25,6 +26,20 @@ config.integrationsList.forEach((org) => {
     integrations: { github },
     run: async (payload, io, ctx) =>
       createIssueCreationJob<IOWithIntegrations<{ github: Autoinvoicing }>>(payload, io, ctx, org)
+  });
+
+  client.defineJob({
+    // This is the unique identifier for your Job, it must be unique across all Jobs in your project
+    id: `issue-comment-streaming_${org.id}${isDev ? '_dev' : ''}`,
+    name: 'Streaming issue comment for Github using app',
+    version: '0.0.1',
+    trigger: github.triggers.org({
+      event: events.onIssueComment,
+      org: org.name
+    }),
+    integrations: { github },
+    run: async (payload, io, ctx) =>
+      createIssueCommentJob<IOWithIntegrations<{ github: Autoinvoicing }>>(payload, io, ctx, org)
   });
 
   client.defineJob({

--- a/src/lib/server/trigger-dev/jobs/issues-comment.ts
+++ b/src/lib/server/trigger-dev/jobs/issues-comment.ts
@@ -43,7 +43,7 @@ export async function createJob<T extends IOWithIntegrations<{ github: Autoinvoi
         break;
       }
 
-      const previousComment = await io.runTask('delete-previous-pr-comment', async () => {
+      const prevComment = await io.runTask('delete-previous-pr-comment', async () => {
         const pr = await getPullRequestByIssue(issue, orgDetails.id, org.name, repository.name, io);
         if (!pr) {
           return;
@@ -66,13 +66,13 @@ export async function createJob<T extends IOWithIntegrations<{ github: Autoinvoi
         return previousComment;
       });
 
-      if (previousComment) {
+      if (prevComment) {
         await io.runTask('reinsert-pr-comment', async () => {
           await createComment(
             orgDetails.id,
             orgName,
             repository.name,
-            previousComment.body,
+            prevComment.body,
             issue.number,
             io
           );

--- a/src/lib/server/trigger-dev/jobs/issues-comment.ts
+++ b/src/lib/server/trigger-dev/jobs/issues-comment.ts
@@ -38,7 +38,7 @@ export async function createJob<T extends IOWithIntegrations<{ github: Autoinvoi
     case 'created': {
       if (excludedAccounts.includes(payload.sender.login)) {
         io.logger.log('current sender for issue comment is excluded', payload);
-        break;
+        return;
       }
 
       const pr = await getPullRequestByIssue(issue, orgDetails.id, org.name, repository.name, io);

--- a/src/lib/server/trigger-dev/jobs/issues-comment.ts
+++ b/src/lib/server/trigger-dev/jobs/issues-comment.ts
@@ -4,7 +4,7 @@ import type { IssueCommentEvent } from '@octokit/webhooks-types';
 
 import {
   getInstallationId,
-  submissionHeaderCommentForPr,
+  submissionHeaderComment,
   getPullRequestByIssue,
   excludedAccounts,
   reinsertComment
@@ -53,7 +53,7 @@ export async function createJob<T extends IOWithIntegrations<{ github: Autoinvoi
             orgDetails.id,
             org.name,
             repository.name,
-            submissionHeaderCommentForPr(pr.id),
+            submissionHeaderComment('Pull Request', pr.id),
             issue.number,
             io
           );

--- a/src/lib/server/trigger-dev/jobs/issues-comment.ts
+++ b/src/lib/server/trigger-dev/jobs/issues-comment.ts
@@ -3,13 +3,11 @@ import type { Autoinvoicing } from '@holdex/autoinvoicing';
 import type { IssueCommentEvent } from '@octokit/webhooks-types';
 
 import {
-  getPreviousComment,
-  deleteComment,
-  createComment,
   getInstallationId,
   submissionHeaderCommentForPR,
   getPullRequestByIssue,
-  excludedAccounts
+  excludedAccounts,
+  reinsertComment
 } from '../utils';
 
 export async function createJob<T extends IOWithIntegrations<{ github: Autoinvoicing }>>(
@@ -43,41 +41,19 @@ export async function createJob<T extends IOWithIntegrations<{ github: Autoinvoi
         break;
       }
 
-      const prevComment = await io.runTask('delete-previous-pr-comment', async () => {
-        const pr = await getPullRequestByIssue(issue, orgDetails.id, org.name, repository.name, io);
-        if (!pr) {
-          return;
-        }
-
-        const previousComment = await getPreviousComment(
-          orgDetails.id,
-          orgName,
-          repository.name,
-          submissionHeaderCommentForPR(pr.id),
-          pr.number,
-          'pullRequest',
-          io
-        );
-
-        if (previousComment) {
-          await deleteComment(orgDetails.id, orgName, repository.name, previousComment, io);
-        }
-
-        return previousComment;
-      });
-
-      if (prevComment) {
-        await io.runTask('reinsert-pr-comment', async () => {
-          await createComment(
-            orgDetails.id,
-            orgName,
-            repository.name,
-            prevComment.body,
-            issue.number,
-            io
-          );
-        });
+      const pr = await getPullRequestByIssue(issue, orgDetails.id, org.name, repository.name, io);
+      if (!pr) {
+        return;
       }
+
+      await reinsertComment(
+        orgDetails.id,
+        org.name,
+        repository.name,
+        submissionHeaderCommentForPR(pr.id),
+        issue.number,
+        io
+      );
 
       break;
     }

--- a/src/lib/server/trigger-dev/jobs/issues-comment.ts
+++ b/src/lib/server/trigger-dev/jobs/issues-comment.ts
@@ -4,7 +4,7 @@ import type { IssueCommentEvent } from '@octokit/webhooks-types';
 
 import {
   getInstallationId,
-  submissionHeaderCommentForPR,
+  submissionHeaderCommentForPr,
   getPullRequestByIssue,
   excludedAccounts,
   reinsertComment
@@ -50,7 +50,7 @@ export async function createJob<T extends IOWithIntegrations<{ github: Autoinvoi
         orgDetails.id,
         org.name,
         repository.name,
-        submissionHeaderCommentForPR(pr.id),
+        submissionHeaderCommentForPr(pr.id),
         issue.number,
         io
       );

--- a/src/lib/server/trigger-dev/jobs/issues-comment.ts
+++ b/src/lib/server/trigger-dev/jobs/issues-comment.ts
@@ -1,0 +1,98 @@
+import type { TriggerContext, IOWithIntegrations } from '@trigger.dev/sdk';
+import type { Autoinvoicing } from '@holdex/autoinvoicing';
+import type { IssueCommentEvent } from '@octokit/webhooks-types';
+
+import {
+  getPreviousComment,
+  deleteComment,
+  createComment,
+  getInstallationId,
+  submissionHeaderCommentForPR,
+  getPullRequestByIssue,
+  excludedAccounts
+} from '../utils';
+
+export async function createJob<T extends IOWithIntegrations<{ github: Autoinvoicing }>>(
+  payload: IssueCommentEvent,
+  io: T,
+  ctx: TriggerContext,
+  org: { nodeId: string; name: string }
+) {
+  const { action, organization, repository, issue } = payload;
+  const orgName = organization?.login || 'holdex';
+
+  const isPullRequest = issue.html_url.includes('/pull/');
+  if (!isPullRequest) {
+    io.logger.log('comment creation in issue is not in the parse candidate', payload);
+    return;
+  }
+
+  const orgDetails = await io.runTask(
+    'get-org-installation',
+    async () => {
+      const { data } = await getInstallationId(orgName);
+      return data;
+    },
+    { name: 'Get Organization installation' }
+  );
+
+  switch (action) {
+    case 'created': {
+      if (excludedAccounts.includes(payload.sender.login)) {
+        io.logger.log('current sender for issue comment is excluded', payload);
+        break;
+      }
+
+      const pr = await io.runTask('convert-issue-to-pr', async () => {
+        const res = await getPullRequestByIssue(
+          issue,
+          orgDetails.id,
+          org.name,
+          repository.name,
+          io
+        );
+        return res;
+      });
+
+      if (!pr) {
+        break;
+      }
+
+      const previousComment = await io.runTask('delete-previous-pr-comment', async () => {
+        const previousComment = await getPreviousComment(
+          orgDetails.id,
+          orgName,
+          repository.name,
+          submissionHeaderCommentForPR(pr.id),
+          pr.number,
+          'pullRequest',
+          io
+        );
+
+        if (previousComment) {
+          await deleteComment(orgDetails.id, orgName, repository.name, previousComment, io);
+        }
+
+        return previousComment;
+      });
+
+      if (previousComment) {
+        await io.runTask('reinsert-pr-comment', async () => {
+          await createComment(
+            orgDetails.id,
+            orgName,
+            repository.name,
+            previousComment.body,
+            issue.number,
+            io
+          );
+        });
+      }
+
+      break;
+    }
+    default: {
+      io.logger.log('current action for issue comment is not in the parse candidate', payload);
+    }
+  }
+}

--- a/src/lib/server/trigger-dev/jobs/issues-comment.ts
+++ b/src/lib/server/trigger-dev/jobs/issues-comment.ts
@@ -19,7 +19,7 @@ export async function createJob<T extends IOWithIntegrations<{ github: Autoinvoi
   const { action, organization, repository, issue } = payload;
   const orgName = organization?.login || 'holdex';
 
-  const isPullRequest = issue.html_url.includes('/pull/');
+  const isPullRequest = !!issue.pull_request;
   if (!isPullRequest) {
     io.logger.log('comment creation in issue is not in the parse candidate', payload);
     return;

--- a/src/lib/server/trigger-dev/jobs/issues-comment.ts
+++ b/src/lib/server/trigger-dev/jobs/issues-comment.ts
@@ -43,22 +43,12 @@ export async function createJob<T extends IOWithIntegrations<{ github: Autoinvoi
         break;
       }
 
-      const pr = await io.runTask('convert-issue-to-pr', async () => {
-        const res = await getPullRequestByIssue(
-          issue,
-          orgDetails.id,
-          org.name,
-          repository.name,
-          io
-        );
-        return res;
-      });
-
-      if (!pr) {
-        break;
-      }
-
       const previousComment = await io.runTask('delete-previous-pr-comment', async () => {
+        const pr = await getPullRequestByIssue(issue, orgDetails.id, org.name, repository.name, io);
+        if (!pr) {
+          return;
+        }
+
         const previousComment = await getPreviousComment(
           orgDetails.id,
           orgName,

--- a/src/lib/server/trigger-dev/jobs/issues-comment.ts
+++ b/src/lib/server/trigger-dev/jobs/issues-comment.ts
@@ -37,7 +37,7 @@ export async function createJob<T extends IOWithIntegrations<{ github: Autoinvoi
   switch (action) {
     case 'created': {
       if (excludedAccounts.includes(payload.sender.login)) {
-        io.logger.log('current sender for issue comment is excluded', payload);
+        io.logger.log(`current sender ${payload.sender.login} for issue comment is excluded`);
         return;
       }
 

--- a/src/lib/server/trigger-dev/jobs/issues-comment.ts
+++ b/src/lib/server/trigger-dev/jobs/issues-comment.ts
@@ -46,13 +46,19 @@ export async function createJob<T extends IOWithIntegrations<{ github: Autoinvoi
         return;
       }
 
-      await reinsertComment(
-        orgDetails.id,
-        org.name,
-        repository.name,
-        submissionHeaderCommentForPr(pr.id),
-        issue.number,
-        io
+      await io.runTask(
+        'reinsert-sticky-comment',
+        async () => {
+          return reinsertComment(
+            orgDetails.id,
+            org.name,
+            repository.name,
+            submissionHeaderCommentForPr(pr.id),
+            issue.number,
+            io
+          );
+        },
+        { name: 'Reinsert sticky comment' }
       );
 
       break;

--- a/src/lib/server/trigger-dev/jobs/issues-creation.ts
+++ b/src/lib/server/trigger-dev/jobs/issues-creation.ts
@@ -37,7 +37,7 @@ export async function createJob<T extends IOWithIntegrations<{ github: Autoinvoi
           orgDetails.id,
           orgName,
           repository.name,
-          submissionHeaderComment(payload.issue.id.toString()),
+          submissionHeaderComment('Issue', payload.issue.id.toString()),
           issue.number,
           'issue',
           io
@@ -51,6 +51,7 @@ export async function createJob<T extends IOWithIntegrations<{ github: Autoinvoi
       if (issue.title.length > MAX_TITLE_LENGTH) {
         await io.runTask('add-issue-title-comment', async () => {
           const commentBody = bodyWithHeader(
+            'Issue',
             `@` +
               payload.sender.login +
               ` please change the title of this issue to make sure the length doesn't exceed ` +

--- a/src/lib/server/trigger-dev/jobs/pull-requests-review.ts
+++ b/src/lib/server/trigger-dev/jobs/pull-requests-review.ts
@@ -14,7 +14,7 @@ import {
   reinsertComment,
   runPrFixCheckRun,
   submissionCheckName,
-  submissionHeaderCommentForPR
+  submissionHeaderCommentForPr
 } from '../utils';
 
 import { EventType, type ItemSchema } from '$lib/@types';
@@ -53,7 +53,7 @@ export async function createJob<T extends IOWithIntegrations<{ github: Autoinvoi
           orgDetails.id,
           orgName,
           repository.name,
-          submissionHeaderCommentForPR(pull_request.id.toString()),
+          submissionHeaderCommentForPr(pull_request.id.toString()),
           pull_request.number,
           io
         );

--- a/src/lib/server/trigger-dev/jobs/pull-requests-review.ts
+++ b/src/lib/server/trigger-dev/jobs/pull-requests-review.ts
@@ -14,7 +14,7 @@ import {
   reinsertComment,
   runPrFixCheckRun,
   submissionCheckName,
-  submissionHeaderCommentForPr
+  submissionHeaderComment
 } from '../utils';
 
 import { EventType, type ItemSchema } from '$lib/@types';
@@ -56,7 +56,7 @@ export async function createJob<T extends IOWithIntegrations<{ github: Autoinvoi
               orgDetails.id,
               orgName,
               repository.name,
-              submissionHeaderCommentForPr(pull_request.id.toString()),
+              submissionHeaderComment('Pull Request', pull_request.id.toString()),
               pull_request.number,
               io
             );

--- a/src/lib/server/trigger-dev/jobs/pull-requests-review.ts
+++ b/src/lib/server/trigger-dev/jobs/pull-requests-review.ts
@@ -11,8 +11,10 @@ import {
   getContributorInfo,
   getInstallationId,
   getPrInfo,
+  reinsertComment,
   runPrFixCheckRun,
-  submissionCheckName
+  submissionCheckName,
+  submissionHeaderCommentForPR
 } from '../utils';
 
 import { EventType, type ItemSchema } from '$lib/@types';
@@ -23,6 +25,7 @@ export async function createJob<T extends IOWithIntegrations<{ github: Autoinvoi
   ctx: TriggerContext
 ) {
   const { action, pull_request, repository, organization, review } = payload;
+  const orgName = organization?.login || 'holdex';
 
   switch (action) {
     case 'submitted': {
@@ -34,10 +37,6 @@ export async function createJob<T extends IOWithIntegrations<{ github: Autoinvoi
         review.state === 'changes_requested' ||
         review.state === 'commented'
       ) {
-        await insertPrEvent(payload, io);
-
-        const prInfo = await updatePrInfo(payload, io, (s) => s);
-
         const orgDetails = await io.runTask(
           'get org installation',
           async () => {
@@ -45,6 +44,18 @@ export async function createJob<T extends IOWithIntegrations<{ github: Autoinvoi
             return data;
           },
           { name: 'Get Organization installation' }
+        );
+
+        await insertPrEvent(payload, io);
+        const prInfo = await updatePrInfo(payload, io, (s) => s);
+
+        await reinsertComment(
+          orgDetails.id,
+          orgName,
+          repository.name,
+          submissionHeaderCommentForPR(pull_request.id.toString()),
+          pull_request.number,
+          io
         );
 
         const contributorList = await io.runTask<any>(

--- a/src/lib/server/trigger-dev/jobs/pull-requests-review.ts
+++ b/src/lib/server/trigger-dev/jobs/pull-requests-review.ts
@@ -49,13 +49,19 @@ export async function createJob<T extends IOWithIntegrations<{ github: Autoinvoi
         await insertPrEvent(payload, io);
         const prInfo = await updatePrInfo(payload, io, (s) => s);
 
-        await reinsertComment(
-          orgDetails.id,
-          orgName,
-          repository.name,
-          submissionHeaderCommentForPr(pull_request.id.toString()),
-          pull_request.number,
-          io
+        await io.runTask(
+          'reinsert-sticky-comment',
+          async () => {
+            return reinsertComment(
+              orgDetails.id,
+              orgName,
+              repository.name,
+              submissionHeaderCommentForPr(pull_request.id.toString()),
+              pull_request.number,
+              io
+            );
+          },
+          { name: 'Reinsert sticky comment' }
         );
 
         const contributorList = await io.runTask<any>(

--- a/src/lib/server/trigger-dev/utils.ts
+++ b/src/lib/server/trigger-dev/utils.ts
@@ -363,7 +363,6 @@ const queryPreviousComment = async <T extends Octokit>(
       `,
       { ...repo, after, number: idNumber }
     );
-    console.log(data);
     // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
     const viewer = data.viewer as UserGQL;
     // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion

--- a/src/lib/server/trigger-dev/utils.ts
+++ b/src/lib/server/trigger-dev/utils.ts
@@ -287,9 +287,9 @@ async function deleteComment(
   repositoryName: string,
   previousComment: any,
   io: any
-): Promise<void> {
+): Promise<boolean> {
   try {
-    await io.runTask('delete-comment', async () => {
+    return await io.runTask('delete-comment', async () => {
       const octokit = await githubApp.getInstallationOctokit(orgID);
       if (previousComment?.databaseId) {
         await octokit.rest.issues.deleteComment({
@@ -297,10 +297,13 @@ async function deleteComment(
           repo: repositoryName,
           comment_id: previousComment.databaseId
         });
+        return true;
       }
+      return false;
     });
   } catch (error) {
     await io.logger.error('delete comment', { error });
+    return (error as { location: string })?.location === 'after_complete_task';
   }
 }
 
@@ -360,6 +363,7 @@ const queryPreviousComment = async <T extends Octokit>(
       `,
       { ...repo, after, number: idNumber }
     );
+    console.log(data);
     // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
     const viewer = data.viewer as UserGQL;
     // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
@@ -464,7 +468,7 @@ async function reinsertComment<T extends IOWithIntegrations<{ github: Autoinvoic
   prOrIssueNumber: number,
   io: T
 ) {
-  const prevComment = await io.runTask('delete-previous-comment', async () => {
+  const { comment, isDeleted } = await io.runTask('delete-previous-comment', async () => {
     const previousComment = await getPreviousComment(
       orgID,
       orgName,
@@ -475,16 +479,17 @@ async function reinsertComment<T extends IOWithIntegrations<{ github: Autoinvoic
       io
     );
 
+    let hasBeenDeleted = false;
     if (previousComment) {
-      await deleteComment(orgID, orgName, repositoryName, previousComment, io);
+      hasBeenDeleted = await deleteComment(orgID, orgName, repositoryName, previousComment, io);
     }
 
-    return previousComment;
+    return { comment: previousComment, isDeleted: hasBeenDeleted };
   });
 
-  if (prevComment) {
+  if (isDeleted && comment) {
     await io.runTask('reinsert-comment', async () => {
-      await createComment(orgID, orgName, repositoryName, prevComment.body, prOrIssueNumber, io);
+      await createComment(orgID, orgName, repositoryName, comment.body, prOrIssueNumber, io);
     });
   }
 }

--- a/src/lib/server/trigger-dev/utils.ts
+++ b/src/lib/server/trigger-dev/utils.ts
@@ -456,6 +456,39 @@ async function getPullRequestByIssue(
   return previousComment;
 }
 
+async function reinsertComment<T extends IOWithIntegrations<{ github: Autoinvoicing }>>(
+  orgID: number,
+  orgName: string,
+  repositoryName: string,
+  header: string,
+  prOrIssueNumber: number,
+  io: T
+) {
+  const prevComment = await io.runTask('delete-previous-comment', async () => {
+    const previousComment = await getPreviousComment(
+      orgID,
+      orgName,
+      repositoryName,
+      header,
+      prOrIssueNumber,
+      'pullRequest',
+      io
+    );
+
+    if (previousComment) {
+      await deleteComment(orgID, orgName, repositoryName, previousComment, io);
+    }
+
+    return previousComment;
+  });
+
+  if (prevComment) {
+    await io.runTask('reinsert-comment', async () => {
+      await createComment(orgID, orgName, repositoryName, prevComment.body, prOrIssueNumber, io);
+    });
+  }
+}
+
 export {
   githubApp,
   runPrFixCheckRun,
@@ -478,5 +511,6 @@ export {
   submissionHeaderComment,
   bodyWithHeaderForPR,
   submissionHeaderCommentForPR,
-  getPullRequestByIssue
+  getPullRequestByIssue,
+  reinsertComment
 };

--- a/src/lib/server/trigger-dev/utils.ts
+++ b/src/lib/server/trigger-dev/utils.ts
@@ -307,20 +307,12 @@ async function deleteComment(
   }
 }
 
-function submissionHeaderComment(header: string): string {
-  return `<!-- Sticky Issue Comment${header} -->`;
+function submissionHeaderComment(type: 'Issue' | 'Pull Request', header: string): string {
+  return `<!-- Sticky ${type} Comment${header} -->`;
 }
 
-function bodyWithHeader(body: string, header: string): string {
-  return `${body}\n${submissionHeaderComment(header)}`;
-}
-
-function submissionHeaderCommentForPr(header: string): string {
-  return `<!-- Sticky Pull Request Comment${header} -->`;
-}
-
-function bodyWithHeaderForPr(body: string, header: string): string {
-  return `${body}\n${submissionHeaderCommentForPr(header)}`;
+function bodyWithHeader(type: 'Issue' | 'Pull Request', body: string, header: string): string {
+  return `${body}\n${submissionHeaderComment(type, header)}`;
 }
 
 const queryPreviousComment = async <T extends Octokit>(
@@ -513,8 +505,6 @@ export {
   createComment,
   bodyWithHeader,
   submissionHeaderComment,
-  bodyWithHeaderForPr,
-  submissionHeaderCommentForPr,
   getPullRequestByIssue,
   reinsertComment
 };

--- a/src/lib/server/trigger-dev/utils.ts
+++ b/src/lib/server/trigger-dev/utils.ts
@@ -312,12 +312,12 @@ function bodyWithHeader(body: string, header: string): string {
   return `${body}\n${submissionHeaderComment(header)}`;
 }
 
-function submissionHeaderCommentForPR(header: string): string {
+function submissionHeaderCommentForPr(header: string): string {
   return `<!-- Sticky Pull Request Comment${header} -->`;
 }
 
-function bodyWithHeaderForPR(body: string, header: string): string {
-  return `${body}\n${submissionHeaderCommentForPR(header)}`;
+function bodyWithHeaderForPr(body: string, header: string): string {
+  return `${body}\n${submissionHeaderCommentForPr(header)}`;
 }
 
 const queryPreviousComment = async <T extends Octokit>(
@@ -509,8 +509,8 @@ export {
   createComment,
   bodyWithHeader,
   submissionHeaderComment,
-  bodyWithHeaderForPR,
-  submissionHeaderCommentForPR,
+  bodyWithHeaderForPr,
+  submissionHeaderCommentForPr,
   getPullRequestByIssue,
   reinsertComment
 };

--- a/src/lib/server/trigger-dev/utils.ts
+++ b/src/lib/server/trigger-dev/utils.ts
@@ -305,6 +305,14 @@ function bodyWithHeader(body: string, header: string): string {
   return `${body}\n${submissionHeaderComment(header)}`;
 }
 
+function submissionHeaderCommentForPR(header: string): string {
+  return `<!-- Sticky Pull Request Comment${header} -->`;
+}
+
+function bodyWithHeaderForPR(body: string, header: string): string {
+  return `${body}\n${submissionHeaderCommentForPR(header)}`;
+}
+
 const queryPreviousComment = async <T extends Octokit>(
   repo: { owner: string; repo: string },
   idNumber: number,
@@ -435,5 +443,7 @@ export {
   deleteComment,
   createComment,
   bodyWithHeader,
-  submissionHeaderComment
+  submissionHeaderComment,
+  bodyWithHeaderForPR,
+  submissionHeaderCommentForPR
 };

--- a/src/lib/server/trigger-dev/utils.ts
+++ b/src/lib/server/trigger-dev/utils.ts
@@ -7,9 +7,15 @@ import type {
   Organization,
   Repository,
   PullRequestEvent,
-  PullRequestReviewEvent
+  PullRequestReviewEvent,
+  Issue
 } from '@octokit/webhooks-types';
-import type { User as UserGQL, Repository as RepoGQL, IssueComment } from '@octokit/graphql-schema';
+import type {
+  User as UserGQL,
+  Repository as RepoGQL,
+  IssueComment,
+  PullRequest as PullRequestGQL
+} from '@octokit/graphql-schema';
 import type { ContributorSchema, ItemSchema } from '$lib/@types';
 import type { IOWithIntegrations } from '@trigger.dev/sdk';
 
@@ -32,7 +38,8 @@ const excludedAccounts: string[] = [
   'coderabbitai',
   'github-advanced-security[bot]',
   'dependabot[bot]',
-  'pr-time-tracker'
+  'pr-time-tracker',
+  'pr-time-tracker[bot]'
 ];
 
 const getContributorInfo = (user: User): Omit<ContributorSchema, 'role' | 'rate'> => ({
@@ -424,6 +431,31 @@ async function createComment(
   }
 }
 
+async function getPullRequestByIssue(
+  issue: Issue,
+  orgID: number,
+  orgName: string,
+  repositoryName: string,
+  io: any
+): Promise<PullRequestGQL | undefined> {
+  const previousComment = await io.runTask('get-pull-request-by-issue', async () => {
+    try {
+      const octokit = await githubApp.getInstallationOctokit(orgID);
+      const data = await octokit.rest.pulls.get({
+        owner: orgName,
+        repo: repositoryName,
+        pull_number: issue.number
+      });
+      return data.data;
+    } catch (error) {
+      await io.logger.error('get pull request id by issue id', { error });
+      return undefined;
+    }
+  });
+
+  return previousComment;
+}
+
 export {
   githubApp,
   runPrFixCheckRun,
@@ -445,5 +477,6 @@ export {
   bodyWithHeader,
   submissionHeaderComment,
   bodyWithHeaderForPR,
-  submissionHeaderCommentForPR
+  submissionHeaderCommentForPR,
+  getPullRequestByIssue
 };

--- a/src/routes/api/github/auth/profile/+server.ts
+++ b/src/routes/api/github/auth/profile/+server.ts
@@ -46,5 +46,6 @@ export const GET: RequestHandler = async ({ cookies }) => {
   })
     .then((r) => r.json())
     .then((r) => (user = r));
+
   return json({ user }, { status: 200 });
 };

